src/db/tables/subscribers.js

```diff
@@ -271,6 +272,59 @@ async function deleteResolutionsWithEmail (id, email) {
 }
 /* c8 ignore stop */
 
+/**
+ * @param {Partial<{ plusOnly: boolean; limit: number; }>} options
+ * @returns {Promise<import("knex/types/tables").SubscriberRow[]>}
+ */
+// Not covered by tests; mostly side-effects. See test-coverage.md#mock-heavy
+/* c8 ignore start */
+async function getSubscribersWaitingForMonthlyEmail (options = {}) {
+  // I'm explicitly referencing the type here, so that these lines of code will
+  // show up as errors when we remove it from the flag list:
+  /** @type {import("./featureFlags.js").FeatureFlagName} */
+  const featureFlagName = "MonthlyActivityEmail";
+  const flag = await knex("feature_flags")
+    .first()
+    .where("name", featureFlagName);
+
+  if (!flag?.is_enabled) {
+    return [];
+  }
+
+  let query = knex('subscribers')
+    .select()
+    .where((builder) => builder.whereNull("monthly_email_optout").orWhere("monthly_email_optout", false))
+    .andWhere(builder => builder.whereNull("monthly_email_at").orWhereRaw('"monthly_email_at" < NOW() - INTERVAL \'30 days\''));
```

Comment:
So this avoids re-sending because `updateMonthlyEmailTimestamp()` is called which sets this, right? I just want to make sure we're thinking through that path if there are any exceptions, so we can't e.g. send duplicate emails because something there fails after the email is sent but before this timestamp is updated.

Reply:
Yes exactly, and that gets set [_before_ we send the email](https://github.com/mozilla/blurts-server/blob/3f63a82c31aacc635b99b88dd10163ee00a543ae/src/cronjobs/monthlyActivity.tsx#L46-L48), so if an error occurs, the email won't get sent. Although given https://github.com/mozilla/blurts-server/pull/4399#discussion_r1556466715, I'll probably need to create a new query once that lands.

Reply:
Thinking through this more, what if we:
1) drop the columns (and data) of `monthly_email_optout` and `monthly_email_at`
2) add new cols `monthly_monitor_report: boolean` and `monthly_monitor_report_at:timestamp`

Reply:
Also I believe the data that exists should be inconsequential (very outdated, at least 6-12 months old)

Reply:
Yeah that works for me. Wouldn't really change the logic here, just different column names, and a new query to update the timestamp when I send an email.

Reply:
OK, [rebased](https://github.com/mozilla/blurts-server/pull/4400#discussion_r1576104457) on latest `main` that has the changes described above, and added e34b2e57b to make use of them.